src/test/java/org/broadinstitute/consent/http/service/dao/VoteServiceDAOTest.java

```diff
@@ -118,4 +131,106 @@ public void testUpdateVotesWithValue_closedElection() throws Exception {
 
     serviceDAO.updateVotesWithValue(List.of(vote), true, "rationale");
   }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testUpdateVotesWithValue_MultipleElectionsDifferentStatuses() throws Exception {
+    User user = createUser();
+    DataAccessRequest dar = createDataAccessRequestV3();
+    DataSet dataset = createDataset();
+    Election openElection = createAccessElection(dar.getReferenceId(), dataset.getDataSetId());
+    Election canceledElection = createCancelledAccessElection(dar.getReferenceId(), dataset.getDataSetId());
+    Vote vote1 = createDacVote(user.getDacUserId(), openElection.getElectionId());
+    Vote vote2 = createDacVote(user.getDacUserId(), canceledElection.getElectionId());
+    String rationale = "rationale";
+    initService();
+
+    serviceDAO.updateVotesWithValue(List.of(vote1, vote2), true, rationale);
+  }
+
+  @Test
+  public void testUpdateVotesWithValue_OpenRPElection() throws Exception {
+    testUpdateVotesWithValue_RPElectionWithStatus(ElectionStatus.OPEN);
+  }
+
+  @Test
+  public void testUpdateVotesWithValue_ClosedRPElection() throws Exception {
+    testUpdateVotesWithValue_RPElectionWithStatus(ElectionStatus.CLOSED);
+  }
+
+  @Test
+  public void testUpdateVotesWithValue_CanceledRPElection() throws Exception {
+    testUpdateVotesWithValue_RPElectionWithStatus(ElectionStatus.CANCELED);
+  }
+
+  @Test
+  public void testUpdateVotesWithValue_FinalRPElection() throws Exception {
+    testUpdateVotesWithValue_RPElectionWithStatus(ElectionStatus.FINAL);
+  }
+
+  @Test
+  public void testUpdateVotesWithValue_PendingApprovalRPElection() throws Exception {
+    testUpdateVotesWithValue_RPElectionWithStatus(ElectionStatus.PENDING_APPROVAL);
+  }
+
+  @Test
+  public void testUpdateVotesWithValue_MultipleRPElections() throws Exception {
+    User user = createUser();
+    DataAccessRequest dar = createDataAccessRequestV3();
+    DataSet dataset = createDataset();
+    Election election1 = createRPElection(dar.getReferenceId(), dataset.getDataSetId());
+    Election election2 = createRPElection(dar.getReferenceId(), dataset.getDataSetId());
+    Election election3 = createRPElection(dar.getReferenceId(), dataset.getDataSetId());
+    changeElectionStatus(election2, ElectionStatus.CANCELED);
+    changeElectionStatus(election3, ElectionStatus.CLOSED);
+    Vote vote1 = createDacVote(user.getDacUserId(), election1.getElectionId());
+    Vote vote2 = createDacVote(user.getDacUserId(), election2.getElectionId());
+    Vote vote3 = createDacVote(user.getDacUserId(), election3.getElectionId());
+    String rationale = "rationale";
+    initService();
+
+    List<Vote> votes = serviceDAO.updateVotesWithValue(List.of(vote1, vote2, vote3), true, rationale);
+    assertNotNull(votes);
+    assertFalse(votes.isEmpty());
```

Comment:
nit: You could also check on the ids to ensure that the votes being updated match those in the request list. I'm guessing there might be other tests where you could run this check as well.

Reply:
When I tried to re-run the tests locally, I got an error `java.lang.IllegalArgumentException: resource consent-config.yml not found.` (part of the setup in the test helper class) even though I can see a `consent-config.yml ` file in my resource package. This is the case even with the code the exact same as here, so I'm not sure what's wrong.

Reply:
The tests work fine for me locally, so I wonder if the tests didn't compile correctly on your end. You can try running `mvn test-compile` on the CLI and see if that helps (though I think IntelliJ already does that for you whenever you run a test locally)

Reply:
After running `mvn test-compile` and trying to run the tests, I get `java.io.IOException: Failed to bind to 0.0.0.0/0.0.0.0:8180` due to bind already being in use

Reply:
That means you're running the app at the same time as tests. Quit docker and re-run them.